src/datasets/load.py

```diff
@@ -1308,6 +1314,130 @@ def get_module(self) -> DatasetModule:
         )
 
 
+class HubDatasetModuleFactoryWithParquetExport(_DatasetModuleFactory):
+    """
+    Get the module of a dataset loaded from parquet files of a dataset repository parquet export.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        from_sha: Optional[str] = None,
+        download_config: Optional[DownloadConfig] = None,
+    ):
+        self.name = name
+        self.from_sha = from_sha
+        self.download_config = download_config or DownloadConfig()
+        increase_load_count(name, resource_type="dataset")
+
+    def _get_exported_parquet_files(self) -> List[Dict[str, Any]]:
```

Comment:
Maybe let's define a private `utils/_datasets_server.py` module and put the logic from the `_get_exported_parquet_files` and `_get_exported_dataset_infos` (and `DatasetsServerError`) there.

IIRC our goal is to have an API to interact with `datasets-server` in `huggingface_hub` eventually, so I guess this is a temporary solution